doc/manual/src/language/identifiers.md

```diff
@@ -0,0 +1,50 @@
+# Identifiers
+
+An *identifier* is an [ASCII](https://en.wikipedia.org/wiki/ASCII) character sequence that:
+- Starts with a letter (`a-z`, `A-Z`) or underscore (`_`)
+- Can contain any number of:
+  - Letters (`a-z`, `A-Z`)
+  - Digits (`0-9`)
+  - Underscores (`_`)
+  - Apostrophes (`'`)
+  - Hyphens (`-`)
+- Is not one of the [keywords](#keywords)
+
+> **Syntax**
+>
+> *identifier* ~ `[A-Za-z_][A-Za-z0-9_'-]*`
+
+# Names
+
+A name can be an [identifier](#identifier) or a [string literal](./syntax.md#string-literal).
+
+> **Syntax**
+>
+> *name* → *identifier* | *string*
+
+Names are used in [attribute sets](./syntax.md#attrs-literal), [`let` bindings](./syntax.md#let-expressions), and [`inherit`](./syntax.md#inheriting attributes).
```

Comment:
I don't know if this section should stay on this page but it's cool for now.

Ultimately that last sentence should be more like
```
Names are used in [attribute paths] and [`inherit`] bindings.
```

And attribute paths should be documented as being usable with attribute sets, let-expressions, selectors, and the `?` operator.

I had figured that attribute paths would be documented on a page that covered bindings in general, including `inherit`; I think this section would end up on that page as well.

All of that is probably out of scope for this fragment-of-a-fragment, but I wanted to share in case you would prefer to leave this with the to-be-reworked attribute set stuff for now, given that intent.